sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java

```diff
@@ -207,40 +215,57 @@ public UnmodifiableList<URI> getApplicableWriteEndpoints(RxDocumentServiceReques
         return this.getApplicableWriteEndpoints(request.requestContext.getExcludeRegions());
     }
 
-    public UnmodifiableList<URI> getApplicableWriteEndpoints(List<String> excludedRegions) {
+    public UnmodifiableList<URI> getApplicableWriteEndpoints(List<String> excludedRegionsOnRequest) {
+
         UnmodifiableList<URI> writeEndpoints = this.getWriteEndpoints();
+        Supplier<CosmosExcludedRegions> excludedRegionsSupplier = this.connectionPolicy.getExcludedRegionsSupplier();
 
-        if (excludedRegions == null ||
-            excludedRegions.isEmpty()) {
+        List<String> effectiveExcludedRegions = excludedRegionsSupplier != null ? new ArrayList<>(excludedRegionsSupplier
```

Comment:
Here we are converting set to a list, which should be fine, but can't we use the underlying set here directly?

Reply:
We would have to do one round of conversion anyway (either from `Set` to `List` or vice-versa) considering the `excludedRegions` from `RequestOptions` happens to be a `List`.